Source/Core/Core/HW/MMIO.cpp

```diff
@@ -195,6 +195,105 @@ WriteHandlingMethod<T>* InvalidWrite()
   });
 }
 
+// Debug: specialization of the complex handling type which acts as wrapper for all other handler.
+// It will additionally show each memory access.
+template <typename T>
+class DebugHandlingMethod : public ReadHandlingMethod<T>,
```

Comment:
I'd rather not have handlers that do both reads and writes (or visitors that do both reads and writes). As you've likely noticed, the interface is not designed for this, and you end up having to disambiguate based on the "const" in the address which is easy to miss.

Why not have DebugRead/DebugWrite be subclasses of ComplexRead/ComplexWrite and ReadVisitor/WriteVisitor respectively?